File: code/back-end/TellYourFriends/TellYourFriends/Models/Entity/Book.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace TellYourFriends.Models.Entity
{
    public class Book
    {
        [Key]
        public int Id { get; set; }

        public string Name { get; set; }

        public string Author { get; set; }

        public string Description { get; set; }

        public string Edition { get; set; }

        public string Image { get; set; }

        public double Rating { get; set; }

        [ForeignKey("User")]
        public int UserId { get; set; }

        public User User { get; set; }

        public List<Category> Categories { get; set; }

        public List<Comment> Comments { get; set; }
    }
}

File: code/front-end/src/app/models/book.ts
import {Category} from './category';
import {Comment} from './comment';
import {User} from './user';
export class Book {
  id?: number;
  userId?: number;
  name: string;
  author: string;
  description: string;
  edition: string;
  image?: string;
  rating: number;
  user?: User;
  categories?: Array<Category>;
  comments?: Array<Comment>;

  constructor(item: any) {
    if (typeof(item) === 'number') {
      this.id = item;
    } else {
      this.id = item.Id;
      this.userId = item.UserId;
      this.name = item.Name;
      this.author = item.Author;
      this.description = item.Description;
      this.edition = item.Edition;
      this.image = item.Image;
      this.rating = item.Rating;
      this.user = item.User;
      this.categories = item.Categories;
      this.comments = item.Comments;
    }
  }
}
